### PR TITLE
Taxonomy fields not available for selection if they have the same name as a content type

### PR DESCRIPTION
**Pull request by agentrickard**
https://www.drupal.org/project/workbench_access/issues/3019231

### CODE REVIEW

**Comment by agentrickard** [comment]
@larowlan This could use a quick review. Then we can roll a new bugfix release.

File: src/Plugin/AccessControlHierarchy/Taxonomy.php [hunk 1]
```diff
@@ -328,7 +329,11 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
             $handler_settings = $field_definitions[$field_name]->getSetting('handler_settings');
             // Must refer to a proper target. Target bundles referring to
             // themselves would create an infinite loop. Deny.
-            if (!isset($handler_settings['target_bundles']) || in_array($bundle, $handler_settings['target_bundles'], TRUE)) {
+            if ($entity_type_id == 'taxonomy_term' && in_array($bundle, $this->configuration['vocabularies'])) {
```
Comment (larowlan) [inline src/Plugin/AccessControlHierarchy/Taxonomy.php]: we should always use the third argument to in_array when comparing strings https://3v4l.org/vnfLY

File: src/Plugin/AccessControlHierarchy/Taxonomy.php [hunk 2]
```diff
@@ -328,7 +329,11 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
...
+              continue;
+            }
+            $allowed = array_intersect($handler_settings['target_bundles'], $this->configuration['vocabularies']);
+            if (!isset($handler_settings['target_bundles']) || empty($allowed)) {
```
Comment (larowlan) [inline src/Plugin/AccessControlHierarchy/Taxonomy.php]: we've already accessed $handler_settings['target_bundles'] without the isset check. So either its always set (and we don't need it) or we risk triggering a warning here by accessing it without checking first.

**Review by larowlan — COMMENTED** [review]
not sure I understand the problem we're fixing, but reviewed the code

**Comment by agentrickard** [comment]
The problem here is "I created a content type named 'Foo' and a vocabulary 'Foo' and now I cannot use Vocabulary 'Foo' fields for access control because of namespace collisions."

The other issues are the review I needed.

**Pull request merged**